### PR TITLE
feat: added ad auth to the stac-browser

### PR DESCRIPTION
**Pull request by ivica3730k**
Inspected a bit around the stac-browser and realised it is using Axios to send requests to the stac-fastapi.

Hence, inherited the ad-auth component from the stac-portal frontend and just reused it for the stac browser.

### CODE REVIEW

File: src/auth/auth.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Modules
+import axios from "axios";
+
+const getAADToken = async () => {
```
Comment (james-hinton) [inline src/auth/auth.js:4]: it'd be good to check if we are in localhost, as when working with it locally we get spammed with network requests to /auth/.me that constantly fail and throw errors.

Comment (ivica3730k) [reply src/auth/auth.js:4]: Bear in mind that code would also need to work for a build version. Can we instead do: do not send it if it is not a build? 

Is there any env var that node server exposes when it is running?

``` js
let isDevServer = process.env.NODE_SOMETHING_DEV || false;

if {!isDevServer}{
  auth();
}
```

Comment (james-hinton) [reply src/auth/auth.js:4]: would work

```
if (process.env.NODE_ENV === 'production') {
}
```

Comment (ivica3730k) [reply src/auth/auth.js:4]: you don't have environment variables when you do build. It is just static HTML, CSS and JS

Comment (ivica3730k) [reply src/auth/auth.js:4]: I can put enable-auth as a build option, I will do that (:like)

Comment (james-hinton) [reply src/auth/auth.js:4]: We will have node_env set before build. its a necessary environment variable to have

Comment (ivica3730k) [reply src/auth/auth.js:4]: Thank you for your input, I will code it in and let you know. 

If it works and feels okay, we can do it the same way on the frontend too if you would like!

Comment (james-hinton) [reply src/auth/auth.js:4]: we could use the VUE_APP_ env if you'd prefer. but the NODE_ENV would get added on build im pretty sure

**Comment by ivica3730k** [comment]
Sorry, I will merge this in for now and open another issue for the above. Need to proceed with deployment

**Pull request merged**